include/devicetree/pwms.h

```diff
@@ -101,6 +101,79 @@ extern "C" {
  */
 #define DT_PWMS_LABEL(node_id) DT_PWMS_LABEL_BY_IDX(node_id, 0)
 
+/**
+ * @brief Get a pwm-controller node identifier from a pwms property at an index
+ *
+ * Example devicetree fragment:
+ *
+ *     pwm1: pwm-controller@... {
+ *             label = "PWM_1";
+ *     };
+ *
+ *     pwm2: pwm-controller@... {
+ *             label = "PWM_2";
+ *     };
+ *
+ *     n: node {
+ *             pwms = <&pwm1 1 PWM_POLARITY_NORMAL>,
+ *                    <&pwm2 3 PWM_POLARITY_INVERTED>;
+ *     };
+ *
+ * Example usage:
+ *
+ *     DT_PWMS_CTLR_BY_IDX(DT_NODELABEL(n), 0) // "PWM_1"
+ *     DT_PWMS_CTLR_BY_IDX(DT_NODELABEL(n), 1) // "PWM_2"
+ *
+ * @param node_id node identifier for a node with a pwms property
+ * @param idx logical index into pwms property
+ * @return  The node identifier for the node referenced at index "idx"
+ * @see DT_PROP_BY_PHANDLE_IDX()
+ */
+#define DT_PWMS_CTLR_BY_IDX(node_id, idx) \
+	DT_PHANDLE_BY_IDX(node_id, pwms, idx)
+
+/**
+ * @brief Get a pwm-controller node identifier from a pwms property by name
+ *
+ * Example devicetree fragment:
+ *
+ *     pwm1: pwm-controller@... {
+ *             label = "PWM_1";
+ *     };
+ *
+ *     pwm2: pwm-controller@... {
+ *             label = "PWM_2";
+ *     };
+ *
+ *     n: node {
+ *             pwms = <&pwm1 1 PWM_POLARITY_NORMAL>,
+ *                    <&pwm2 3 PWM_POLARITY_INVERTED>;
+ *             pwm-names = "alpha", "beta";
+ *     };
+ *
+ * Example usage:
+ *
+ *     DT_PWMS_CTLR_BY_NAME(DT_NODELABEL(n), alpha) // "PWM_1"
+ *     DT_PWMS_CTLR_BY_NAME(DT_NODELABEL(n), beta)  // "PWM_2"
+ *
+ * @param node_id node identifier for a node with a pwms property
+ * @param name lowercase-and-underscores name of a pwms element
+ *             as defined by the node's pwm-names property
+ * @return The node identifier to the pwm-controller for the named element
+ * @see DT_PHANDLE_BY_NAME()
+ */
+#define DT_PWMS_CTLR_BY_NAME(node_id, name) \
+	DT_PHANDLE_BY_NAME(node_id, pwms, name)
+
+/**
+ * @brief Equivalent to DT_PWMS_CTLR_BY_IDX(node_id, 0)
+ * @param node_id node identifier for a node with a pwms property
+ * @return The node identifier of the pwm-controller of the node
+ *         referenced at index 0
```

Comment:
```suggestion
 * @return the node identifier for the PWM controller at index 0
 *         in the node's "pwms" property
```

This is closer to how DT_GPIO_CTLR is written.